client/src/components/PublicPages/AuditPlanner/sampleSizes.ts

```diff
@@ -0,0 +1,58 @@
+import { toast } from 'react-toastify'
+import { useQuery, UseQueryResult } from 'react-query'
+
+import { fetchApi } from '../../../utils/api'
+import { IElectionResults } from './electionResults'
+import { sleep } from '../../../utils/sleep'
+
+export type AuditType = 'ballotComparison' | 'ballotPolling' | 'batchComparison'
+
+export type SampleSizes = { [key in AuditType]: number }
+
+export const placeholderSampleSizes: SampleSizes = {
+  ballotComparison: 0,
+  ballotPolling: 0,
+  batchComparison: 0,
+}
+
+interface UseSampleSizesOptions {
+  /** Allow introducing an artificial delay to avoid flickering loading spinners */
+  minFetchDurationMs?: number
+  /** Allow overriding the global onError behavior of showing a toast */
+  showToastOnError?: boolean
+}
+
+export const useSampleSizes = (
+  electionResults: IElectionResults,
+  riskLimitPercentage: number,
+  {
+    minFetchDurationMs = 0,
+    showToastOnError = true,
+  }: UseSampleSizesOptions = {}
+): UseQueryResult<SampleSizes, Error> =>
+  useQuery<SampleSizes, Error>(
```

Comment:
Ah yes I meant to drop an explanatory comment! Only reason I'm using POST is so that I can specify election results in a body. Conceptually, this is more of a GET since it's just fetching data and not actually creating anything. But if I was to try specifying election results in a GET query param, I may hit URL size limits

Reply:
Cool, makes sense. That was my hunch